pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -105,66 +106,80 @@ private static String removeTerminatingSemicolon(String sql) {
   }
 
   public static SqlNodeAndOptions compileToSqlNodeAndOptions(String sql)
-      throws Exception {
+      throws SqlCompilationException {
     // Remove the comments from the query
     sql = removeComments(sql);
 
     // Remove the terminating semicolon from the query
     sql = removeTerminatingSemicolon(sql);
 
-    // Extract OPTION statements from sql as Calcite Parser doesn't parse it.
+    // extract and remove OPTIONS string
     List<String> options = extractOptionsFromSql(sql);
-    if (!options.isEmpty()) {
-      sql = removeOptionsFromSql(sql);
-    }
+    sql = removeOptionsFromSql(sql);
```

Comment:
Don't remove the if check to avoid overhead when there is no options

Reply:
```suggestion
    if (!options.isEmpty()) {
      sql = removeOptionsFromSql(sql);
    }
```